save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -25,21 +27,38 @@ data class Warning(
  * @param lineGroupIdx index of capture group for line number
  * @param messageGroupIdx index of capture group for waring text
  * @return a [Warning] or null if [this] string doesn't match [warningRegex]
+ * @throws ResourceFormatException when parsing a file
  */
 internal fun String.extractWarning(warningRegex: Regex,
                                    columnGroupIdx: Int?,
                                    lineGroupIdx: Int?,
                                    messageGroupIdx: Int,
 ): Warning? {
     val groups = warningRegex.find(this)?.groups ?: return null
+
     val line = lineGroupIdx?.let {
-        groups[lineGroupIdx]?.value?.toInt()
+        try {
+            groups[lineGroupIdx]!!.value.toInt()
+        } catch (e: Exception) {
+            throw ResourceFormatException("Invalid line format in test file: $this")
+        }
     }
+
     val column = columnGroupIdx?.let {
-        groups[columnGroupIdx]?.value?.toInt()
+        try {
+            groups[columnGroupIdx]!!.value.toInt()
+        } catch (e: Exception) {
+            throw ResourceFormatException("Invalid column format in test file: $this")
+        }
+    }
+
+    val message = try {
+        groups[messageGroupIdx]!!.value
+    } catch (e: Exception) {
+        throw ResourceFormatException("Invalid warning message in test file: $this")
     }
     return Warning(
```

Comment:
Please also add tests in `warn-plugin`: if test file has invalid lines with warnings, use `assertFailsWith<ResourceFormatException>`